### ISSUE TITLE
Convert code to ES6 and add a transpile build step

### ISSUE THREAD

**Issue by BehindTheMath** [MoOx/pjax#109]

**Comment by robinnorth** [comment]
This sounds sensible, if this still ends up using Browserify with the addition of the Babelify transform, you could add the following to`package.json` to allow people bundling the npm package through Browserify (my usage) to get transpiled output as well (Browserify won't transform anything under `node_modules` by default):

```json
{
  "browserify": {
    "transform": ["babelify"]
  }
}
```

**Comment by MoOx** [comment]
Another possibility is to include in npm package compiled files (but ignore those in git).

**Comment by phifa** [comment]
Webpack?

**Comment by BehindTheMath** [comment]
@phifa I believe Webpack is overkill for a library like Pjax. Webpack is more suited for a webapp with different types of assets or multiple entry points. For Pjax, Babel and Browserify should be fine.

**Comment by MoOx** [comment]
Rollup should be a good candidate.

**Comment by BehindTheMath** [comment]
Do we even need Rollup?

**Comment by MoOx** [comment]
I just know that Rollup is nice for library authors

**Comment by BehindTheMath** [comment]
True. I'll have to look into the differences between Rollup and Browserify.

**Comment by BehindTheMath** [comment]
@MoOx What browserlist settings do you think we should use for Babel?

**Comment by MoOx** [comment]
No idea what is best to use. Keep defaults?
